lib/bolt/resource_instance.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+require 'bolt/target'
+require 'json'
+
+module Bolt
+  class ResourceInstance
+    attr_reader :target, :type, :title, :state, :desired_state
+    attr_accessor :events
+
+    def initialize(resource_hash)
+      @target        = resource_hash['target']
+      @type          = resource_hash['type']
+      @title         = resource_hash['title']
+      @state         = resource_hash['state'] || resource_hash['parameters'] || {}
```

Comment:
`resource_hash['parameters']` is here so you can create a `ResourceInstance` from the hash returned from `get_resources()`.